src/hostnet/slirp.ml

```diff
@@ -1058,6 +1060,25 @@ module Make(Config: Active_config.S)(Vmnet: Sig.VMNET)(Dns_policy: Sig.DNS_POLIC
                     let ip, _ = v in
                     l @ [ip]) t.client_uuids.table []
             in 
+
+            (* check if a specific IP is requested *)
+            let preferred_ip = 
+                let uuid_bytes = Uuidm.to_bytes uuid in
+                let uuid_prefix = Bytes.sub uuid_bytes 0 (Bytes.length default_uuid_preferred_ip_prefix) in
+                if (Bytes.compare uuid_prefix default_uuid_preferred_ip_prefix) = 0 then
+                begin
+                    let uuid_suffix = Bytes.sub uuid_bytes 12 4 in
+                    let preferred_ip = Ipaddr.V4.of_bytes_exn uuid_suffix in
+                    Log.info (fun f -> f "Client requested IP %s" (Ipaddr.V4.to_string preferred_ip));
```

Comment:
Rather than encode the preferred IP within the UUID, perhaps it would be simpler if we extended the vmnet protocol a little -- we could add a new command type e.g.
```
module Command : sig

  type t =
    | Ethernet of string (* 36 bytes *)
    | Bind_ipv4 of Ipaddr.V4.t * int * bool
    | Preferred_ipv4 of string (* 36 byte uuid *) * Ipaddr.V4.t (** <-- NEW *)
```
Separately we could have a hyperkit patch to add a new key=value pair to the `virtio-vpnkit` driver, so we could have command-lines like this:
```
-s 2:0,virtio-vpnkit,uuid=...,path=...,macfile=....,preferred-ip=...
```
which would then send the new message.